docs/rfcs/20220209_consumer_offsets.md

```diff
@@ -0,0 +1,190 @@
+will alter properties of an aliased topic.
+
+### `CreateTopic`
+
+When create topic is called for an aliased topic we will create aliased internal
+topic with properties requested during topic creation. 
+
+### `Fetch`
+
+Fetch is going to create a dedicated translating reader that will perform data 
+translation on the fly.
+
+### `Produce`
+
+Producing to topic alias will result in producing to aliased `kafka_internal` 
+topic, before persisting data to the internal topic alias related data 
+transformation is going to be applied.
+
+
+### Compatibility with Kafka
+
+Currently we are going to support following version of `__consumer_offsets` types
+
+- `OffsetCommitValue`: version 1 
+- - we do not support the expiryTimestamp but we will always return `-1` i.e. no expiration
+
+- `GroupMetadataValue`: version 3
+
+- `MemberMetadata`: version 3
+- 
+# Guide-level explanation
+## How do we teach this?
+
+- explain to users that we added another layer of compatibility with Kafka
+- explain what kind of data are accessible by consuming `__consumer_offsets` topic
+
+## Introducing new named concepts.
+
+We will introduced `internal_topic_alias` abstraction that will allow mapping topics
+from `kafka_internal` namespace to topics marked as internal in clients 
+accessible `kafka` namespace.
+
+# Reference-level explanation
+
+## Interaction with other features
+
+### RPK
+
+We may want to add a flag `--internal-topics` to RPK to allow user 
+interacting with `__consumer_offsets` topic. Additionally we may want to add 
+decoding capabilities so that RPK will be able to display human/machine 
+readable content of `__consumer_offsets` topic.
+
+### Admin API
+
+There is no need to expose `__consumer_offsets` alias via Admin API. 
+We may assume that admin API users should be aware of Redpanda internals and 
+we may just operate using `kafka_internal/group` Redpanda specific topic name.
+
+
+## Corner cases dissected by example.
+
+### Topic `kafka/__consumer_offsets` already exists in the cluster
+
+In this case we will just use whatever is there and ignore an alias.
+
+### Topic `kafka/__consumer_offsets` is being created by user
+
+In this case Redpanda will create underlying `kafka_internal/groups` topic with 
+a specified set of options provided by a caller. We may emit WARN log message 
+that the topic being created is an internal topic. 
+
+## Detailed design - What needs to change to get there
+
+### Internal topics
+
+We will introduce a function `bool is_internal_topic(const model::topic&)` 
+that will return true whenever a topic is internal. This will provide 
+compatibility with Kafka internal topics abstraction. Topics are marked as 
+internal in  `MetadataResponse`. 
+
+### Topic aliases
+
+Adding an `std::vector<topic_alias>` to topic table where topic aliases will be 
+registered. We can tolerate the linear lookup const since number of aliases 
+is very limited (currently 2).
```

Comment:
```suggestion
Add an `std::vector<topic_alias>` sequence container to the topic table where topic aliases will be 
registered. We can tolerate the linear lookup const since the number of aliases 
is very limited (currently 2).
```

Reply:
make sure my "sequence container" edit is correct